app/src/main/java/net/gsantner/markor/ui/fsearch/SearchEngine.java

```diff
@@ -429,5 +439,24 @@ private boolean isIgnored(File directory) {
             }
             return false;
         }
+
+        private InputStream getInputStream(File file) throws FileNotFoundException {
+            if (isEncryptedFile(file)) {
+                final byte[] encryptedContext = FileUtils.readCloseStreamWithSize(new FileInputStream(file), (int) file.length());
+                if (encryptedContext.length > JavaPasswordbasedCryption.Version.NAME_LENGTH && _config.password.length > 0) {
+                    final byte[] decrypt = new JavaPasswordbasedCryption(getVersion(encryptedContext), null).decryptBytes(encryptedContext, _config.password.clone());
+                    return new ByteArrayInputStream(decrypt);
+                } else {
+                    return new ByteArrayInputStream(encryptedContext);
+                }
+            } else {
+                return new FileInputStream(file);
+            }
+        }
+    }
+
+
+    private static boolean isEncryptedFile(File file) {
+        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && file.getName().endsWith(JavaPasswordbasedCryption.DEFAULT_ENCRYPTION_EXTENSION);
```

Comment:
also mind encryption for isIgnored function (.*.txt filter at .txt.enc VS .txt file)

Reply:
I would say that the caller should be able to define an exact filter. So if he want to ignore encrypted files he can do it. Or if he want to search only in encrypted file. It's tricky to implement there an automatism. 
BTW Base on variable names it looks like ignoring directories, but it seems that it ignores files...